tensorflow_recommenders_addons/dynamic_embedding/python/ops/redis_table_ops.py

```diff
@@ -0,0 +1,397 @@
+            `self._default_value` will be used.
+          name: A name for the operation (optional).
+
+        Returns:
+          A tensor containing the values in the same shape as `keys` using the
+            table's value type.
+
+        Raises:
+          TypeError: when `keys` do not match the table data types.
+        """
+    with ops.name_scope(
+        name,
+        "%s_lookup_table_find" % self.name,
+        (self.resource_handle, keys, self._default_value),
+    ):
+      keys = ops.convert_to_tensor(keys, dtype=self._key_dtype, name="keys")
+      with ops.colocate_with(self.resource_handle):
+        values = redis_table_ops.tfra_redis_table_find(
+            self.resource_handle,
+            keys,
+            dynamic_default_values
+            if dynamic_default_values is not None else self._default_value,
+        )
+    return values
+
+  def insert(self, keys, values, name=None):
+    """Associates `keys` with `values`.
+
+      Args:
+        keys: Keys to insert. Can be a tensor of any shape. Must match the table's
+          key type.
+        values: Values to be associated with keys. Must be a tensor of the same
+          shape as `keys` and match the table's value type.
+        name: A name for the operation (optional).
+
+      Returns:
+        The created Operation.
+
+      Raises:
+        TypeError: when `keys` or `values` doesn't match the table data
+          types.
+    """
+    with ops.name_scope(
+        name,
+        "%s_lookup_table_insert" % self.name,
+        [self.resource_handle, keys, values],
+    ):
+      keys = ops.convert_to_tensor(keys, self._key_dtype, name="keys")
+      values = ops.convert_to_tensor(values, self._value_dtype, name="values")
+      with ops.colocate_with(self.resource_handle):
+        # pylint: disable=protected-access
+        op = redis_table_ops.tfra_redis_table_insert(self.resource_handle, keys,
+                                                     values)
+    return op
+
+  def import_from_files(self, name=None):
+    """restored from a Redis binary dump files which paths are directory
+       '"'[model_lib_abs_dir]/[model_tag]/[name].rdb'.
+
+      Args:
+        name: A name for the operation (optional).
+
+    """
+
+    with ops.name_scope(name, "%s_lookup_table_export_values" % self.name,
+                        [self.resource_handle]):
+      with ops.colocate_with(self.resource_handle):
+        ndims = self._default_value.get_shape().ndims
+        if ndims == 1:
+          useless_keys = ops.convert_to_tensor(0, dtype=self.key_dtype)
+        else:
+          useless_keys = tf.zeros(shape=(self._default_value.get_shape()[0]),
+                                  dtype=self.key_dtype)
+        useless_vals = tf.zeros(shape=self._default_value.get_shape(),
+                                dtype=self.value_dtype)
+        return redis_table_ops.tfra_redis_table_import(self.resource_handle,
```

Comment:
Why use `useless_keys` and `useless_vals`, maybe you should design the API `tfra_redis_table_import` 's inputs.